pom.xml

```diff
@@ -135,10 +173,184 @@
+                    <consoleOutput>${checkStyle.config.consoleOutput}</consoleOutput>
+                    <maxAllowedViolations>${checkStyle.config.maxAllowedViolations}</maxAllowedViolations>
+                    <enableFilesSummary>${checkStyle.config.enableFilesSummary}</enableFilesSummary>
+                    <enableRulesSummary>${checkStyle.config.enableRulesSummary}</enableRulesSummary>
+                    <enableSeveritySummary>${checkStyle.config.enableSeveritySummary}</enableSeveritySummary>
+                    <headerLocation>${checkStyle.config.headerLocation}</headerLocation>
+                </configuration>
+            </plugin>
+            <!-- Print the change log as release notes...-->
+            <!--Run mvn generate-sources, to use-->
+            <plugin>
+                <groupId>se.bjurr.gitchangelog</groupId>
+                <artifactId>git-changelog-maven-plugin</artifactId>
+                <version>${git-changelog-maven-plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>GenerateGitChangelog</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>git-changelog</goal>
+                        </goals>
+                        <configuration>
+                            <!-- A file on filesystem //-->
+                            <file>CHANGELOG.md</file>
+                            <!--<toRef>master</toRef>-->
+                            <templateFile>${project.basedir}/config/changelog.mustache</templateFile>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <!-- Enforce Maven 3.5.0 -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <version>${maven.enforcer.plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>enforce-maven</id>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireMavenVersion>
+                                    <version>${required.maven.version}</version>
+                                </requireMavenVersion>
+                                <requireJavaVersion>
+                                    <version>${required.java.version}</version>
+                                </requireJavaVersion>
+                            </rules>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!-- Setup spotbugs dependency, would normally only be in reporting section -->
```

Comment:
Is something keeping it out of the reporting section at this time?  Feel free to elaborate within your comment.

Reply:
I had attempted to add it to reporting however while in reporting I was unable to download the dependency to my local repository. So I thought I would declare "dependencies", but you can't do that under reporting, so that's why I moved it to the build section. 
The unfortunate consequence of that is that in order to use it you have to explicitly run "mvn spotbugs:check" yourself.